src/app/components/checkout/checkout.component.html

```diff
@@ -0,0 +1,37 @@
+<div class="checkout_container" *ngIf="showCheckout">
+    <form [formGroup]="checkoutOrderForm">
+        <div class="checkout_container__user_information">
+            <label class="checkout_container__user_information--email">email:</label>
+            <input type="email" name="user_email" formControlName="user_email">
+        </div>
+        <div class="error"
+            *ngIf="userData['user_email'].invalid && (userData['user_email'].dirty || userData['user_email'].touched)">
+            <div *ngIf="userData['user_email'].errors?.['required']">Email required</div>
+            <div *ngIf="userData['user_email'].errors?.['pattern']">Provide valid email address</div>
+        </div>
+    </form>
+    <form [formGroup]="checkoutOrderForm">
```

Comment:
Why are you creating form for each of the inputs? Better would be to just have one form tag and include all inside that. Or maybe this way there are some advantages?

Reply:
Changed